File: 1.Layers/1.3Application/1.3.2MainModule/Application.MainModule/DTOs/Mobile/Cortes/DatosBusquedaCortesDTO.cs
﻿/**
 * DatosBusquedaCortesDTO
 * Permite crear un objeto de respuesta DTO
 * en el cual se incluye los datos de la estación,
 * las ventas encontradas y dependiendo si es corte o anticipo
 * se retornara un objeto con los datos de este si existen 
 * @developer Jorge Omar Tovar Martínez
 * @company Neoteck
 * @date 26/12/2018 01:03
 * @update 26/12/2018 01:03
 */
using System.Collections.Generic;
using Application.MainModule.DTOs.Respuesta;

namespace Application.MainModule.DTOs.Mobile.Cortes
{
    /// <summary>
    /// Objeto con los datos a responder para los corte
    /// o anticipos de un CAlmacenGas
    /// </summary>
    public class DatosBusquedaCortesDTO : RespuestaDto
    {
        /// <summary>
        /// Determina si ya se tiene anticipos
        /// </summary>
        public bool hayAnticipos { get; set; }
        /// <summary>
        /// Determina si existen cortes
        /// </summary>
        public bool hayCortes { get; set; }
        /// <summary>
        /// Determina si hay ventas
        /// </summary>
        public bool hayVentas { get; set; }
        /// <summary>
        /// Objeto que contiene información de los anticipos de la estación
        /// </summary>
        public AnticipoInfoDTO anticipo { get; set; }
        /// <summary>
        /// Objeto que contiene información de los cortes
        /// </summary>
        public CorteInfoDTO corte { get; set; }
        /// <summary>
        /// Objeto con la información de la estación a la que se 
        /// le realiza el corte o anticipo
        /// </summary>
        public EstacionesDto estacion { get; set; }
        /// <summary>
        /// Objeto con los datos de las ventas que se encontraron 
        /// </summary>
        public VentasInfoDTO venta { get; set; }

    }
    #region Clases internas de respuesta
    /// <summary>
    /// Permite crear un objeto dto con los datos de 
    /// corte y su total actual en caso de haber alguno
    /// </summary>
    public class CorteInfoDTO
    {
        /// <summary>
        /// Reprecenta el total de los cortes encontrados
        /// </summary>
        public decimal totalCortes { get; set; }
        /// <summary>
        /// Listado de cortes encontrados
        /// </summary>
        public List<CorteDto> cortes { get; set; }
        /// <summary>
        /// Constructor de clase para inicializar la lista de cortes 
        /// </summary>
        public CorteInfoDTO()
        {
            cortes = new List<CorteDto>();
        }
    }
    /// <summary>
    /// Permite crear un objeto con los datos de 
    /// los anticipos realizados y su total en 
    /// caso de haber alguno
    /// </summary>
    public class AnticipoInfoDTO
    {
        /// <summary>
        /// Suma total de los totales de anticipos
        /// </summary>
        public decimal totalAnticipos { get; set; }
        /// <summary>
        /// Listado de anticipos registrados en la estación
        /// </summary>
        public List<AnticipoDto> anticipos { get; set; }
        /// <summary>
        /// Constructor de clase para inicializar el listado de anticipos
        /// </summary>
        public AnticipoInfoDTO()
        {
            anticipos = new List<AnticipoDto>();
        }
    }
    /// <summary>
    /// Permite crear un objeto con los datos de 
    /// las ventas de una estación y 
    /// </summary>
    public class VentasInfoDTO
    {
        /// <summary>
        /// Reprecenta la suma total de las ventas realizadas
        /// </summary>
        public decimal totalVentas { get; set; }
        /// <summary>
        /// Lista de ventas encontradas para el corte o anticipo
        /// </summary>
        public List<VentaDTO> ventas { get; set; }
        /// <summary>
        /// Constructor de clase para inicializar la lista de ventas
        /// </summary>
        public VentasInfoDTO()
        {
            ventas = new List<VentaDTO>();
        }
    }
    #endregion
}

File: 1.Layers/1.1Presentation/1.1.2Mobile/app/src/main/java/com/example/neotecknewts/sagasapp/Model/DatosBusquedaCortesDTO.java
/*
 * DatosBusquedaCortesDTO
 * Clase para el buscador por fecha de cortes y anticipos
 * en el se puede obtener, las ventas, los cortes , los anticipos
 * y los datos de la estación en la que se realizan estas acciónes
 * @author Jorge Omar Tovar Martínez jorge.tovar@neoteck.com.mx
 * @company Neoteck
 * @date 26/12/2018 5:55 p.m.
 * @update 26/12/2018 5:55 p.m.
 */
package com.example.neotecknewts.sagasapp.Model;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class DatosBusquedaCortesDTO extends RespuestaDTO implements Serializable {

    //region Banderas de datos
    /**
     * Determina si se encontraron anticipos
     */
    @SerializedName("hayAnticipos")
    private boolean hayAnticipos;

    /**
     * Determina si se encontraron cortes
     */
    @SerializedName("hayCortes")
    private boolean hayCortes;

    /**
     * Determina si se encontraron ventas
     */
    @SerializedName("hayVentas")
    private boolean hayVentas;
    //endregion
    //region Clases internas
    /**
     * Datos de los anticipos encontrados en la fecha especificada y en la estación
     * que se tiene en session
     */
    @SerializedName("anticipo")
    public AnticipoInfoDTO anticipo;
    /**
     * Datos de los cortes encontrados en la fecha especificada y en la estación
     * que se tiene en session
     */
    @SerializedName("corte")
    public CorteInfoDTO corte;

    /**
     * Estación de la session actual
     */
    @SerializedName("estacion")
    public EstacionDTO estacion;

    /**
     * Ventas encontradas en la fecha especificada y en la estación
     * que se tiene en session
     */
    @SerializedName("venta")
    public  VentaInfoDTO venta;

    //endregion

    //region Gets y sets
    public boolean isHayAnticipos() {
        return hayAnticipos;
    }

    public void setHayAnticipos(boolean hayAnticipos) {
        this.hayAnticipos = hayAnticipos;
    }

    public boolean isHayCortes() {
        return hayCortes;
    }

    public void setHayCortes(boolean hayCortes) {
        this.hayCortes = hayCortes;
    }

    public boolean isHayVentas() {
        return hayVentas;
    }

    public void setHayVentas(boolean hayVentas) {
        this.hayVentas = hayVentas;
    }

    public AnticipoInfoDTO getAnticipo() {
        return anticipo;
    }

    public void setAnticipo(AnticipoInfoDTO anticipo) {
        this.anticipo = anticipo;
    }

    public CorteInfoDTO getCorte() {
        return corte;
    }

    public void setCorte(CorteInfoDTO corte) {
        this.corte = corte;
    }

    public EstacionDTO getEstacion() {
        return estacion;
    }

    public void setEstacion(EstacionDTO estacion) {
        this.estacion = estacion;
    }

    public VentaInfoDTO getVenta() {
        return venta;
    }

    public void setVenta(VentaInfoDTO venta) {
        this.venta = venta;
    }
    //endregion

    //region Clases internas

    public class CorteInfoDTO implements Serializable {

        /**
         * Contiene el total de los cortes realizados en la fecha y estación
         */
        @SerializedName("totalCortes")
        public double totalCortes;

        /**
         * Contiene el listado de cortes realizados en la fecha y estación
         */
        @SerializedName("cortes")
        public List<CorteDTO> cortes;

        /**
         * Constructor de clase para inicializar la lista de cortes
         */
        public CorteInfoDTO(){
            this.cortes = new ArrayList<>();
        }
    }

    public class AnticipoInfoDTO implements Serializable {
        /**
         * Contiene el total actual de anticipos en esta fecha
         */
        @SerializedName("totalAnticipos")
        private double totalAnticipos;

        /**
         * Listado de anticipos registrados con esa fecha
         */
        @SerializedName("anticipos")
        private List<AnticiposDTO> anticipos;

        /**
         * Constructor de clase para inicializar la lista de
         * anticipos
         */
        public AnticipoInfoDTO(){
            this.anticipos = new ArrayList<>();
        }

        public double getTotalAnticipos() {
            return totalAnticipos;
        }

        public void setTotalAnticipos(double totalAnticipos) {
            this.totalAnticipos = totalAnticipos;
        }

        public List<AnticiposDTO> getAnticipos() {
            return anticipos;
        }

        public void setAnticipos(List<AnticiposDTO> anticipos) {
            this.anticipos = anticipos;
        }
    }

    public class EstacionDTO implements Serializable{
        /**
         * Reprecenta el Id de el CAlmacenGas
         */
        @SerializedName("IdAlmacenGas")
        private int IdCAlmacenGas;

        /**
         * Reprecenta el nombre de la estación,pipa o camioneta
         */
        @SerializedName("NombreAlmacen")
        private String NombreAlmacen;

        /**
         * Reprecenta el porcentaje actual del medidor
         */
        @SerializedName("PorcentajeMedidor")
        private double PorcentajeMedidor;

        public int getIdCAlmacenGas() {
            return IdCAlmacenGas;
        }

        public void setIdCAlmacenGas(int idCAlmacenGas) {
            IdCAlmacenGas = idCAlmacenGas;
        }

        public String getNombreAlmacen() {
            return NombreAlmacen;
        }

        public void setNombreAlmacen(String nombreAlmacen) {
            NombreAlmacen = nombreAlmacen;
        }

        public double getPorcentajeMedidor() {
            return PorcentajeMedidor;
        }

        public void setPorcentajeMedidor(double porcentajeMedidor) {
            PorcentajeMedidor = porcentajeMedidor;
        }
    }

    public class VentaInfoDTO implements Serializable{
        /**
         * Reprecenta la suma total de las ventas en esa fecha y en la estación
         */
        @SerializedName("totalVentas")
        private double totalVentas;

        /**
         * Reprecenta el listado de ventas realizadas en esa fecha y en la estación
         */
        @SerializedName("ventas")
        private List<VentaDTO> ventas;

        /**
         * Constructor de clase para incializar la lista de ventas
         */
        public VentaInfoDTO(){
            this.ventas = new ArrayList<>();
        }

        public double getTotalVentas() {
            return totalVentas;
        }

        public void setTotalVentas(double totalVentas) {
            this.totalVentas = totalVentas;
        }

        public List<VentaDTO> getVentas() {
            return ventas;
        }

        public void setVentas(List<VentaDTO> ventas) {
            this.ventas = ventas;
        }
    }


    //endregion
}
